rust/rpmostree-client/src/lib.rs

```diff
@@ -49,6 +50,7 @@ pub struct Deployment {
     pub pinned: bool,
     pub checksum: String,
     pub base_checksum: Option<String>,
+    #[serde(deserialize_with = "deserialize_base_commit_meta")]
```

Comment:
I think it'd be easier (and more reliable) to just add a method to this `get_base_manifest(&self) -> Result<Option<oci_spec::image::ImageManifest>>` that directly accesses the expected key, instead of trying to heuristically parse every single metadata key as json.

Reply:
Good suggestion ! I made it work that way :)